app/email.py

```diff
@@ -1,21 +1,16 @@
-from threading import Thread
-
-from flask import current_app, render_template
+import os
+from flask import render_template
 from flask.ext.mail import Message
 from . import mail
+from app import create_app
 
 
-def send_async_email(app, msg):
+def send_async_email(recipient, subject, template, **kwargs):
+    app = create_app(os.getenv('FLASK_CONFIG') or 'default')
     with app.app_context():
+        msg = Message(app.config['EMAIL_SUBJECT_PREFIX'] + ' ' + subject,
+                      sender=app.config['EMAIL_SENDER'],
+                      recipients=[recipient])
+        msg.body = render_template(template + '.txt', **kwargs)
+        msg.html = render_template(template + '.html', **kwargs)
         mail.send(msg)
```

Comment:
When you call this function (from wherever), it executes synchronously, right? Before, actually sending the message was relegated to a separate thread which the function `send_email` only started. But now when you call `send_async_email` it creates the app, ravages it for config vars, and then sends the message, but it's still getting executed in the same thread as the caller function. 

When I see `async`, I think, if I call this function, execution will resume before the email has actually been sent. That doesn't seem to the case here?